backend-worker/src/main/kotlin/org/rucca/snake/worker/PermissionService.kt

```
@@ -0,0 +1,86 @@
+        customLogicData: Any?,
+    ): Boolean {
+        val role =
+            (customLogicData as? Map<*, *>)?.get("role") as? String
+                ?: throw RuntimeException(
+                    "Role not found in customLogicData. This is ether a bug or a malicious attack."
+                )
+        val authorization = getAuthorizationForUserWithRole(userId, role)
+        try {
+            authorizationService.audit(authorization, action, resourceType, resourceId, authInfo)
+            return true
+        } catch (e: PermissionDeniedError) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)